sasdata/data_util/nxsunit.py

```diff
@@ -43,173 +43,405 @@
+    sld['10^-6 Angstrom^-2'] = 1e-6
+    DIMENSIONS['sld'] = sld
+
+    # Q units (also inverse lengths)
+    Q = _build_inv_n_metric_units('meter', 'm', 1)
+    Q.update(_build_inv_n_units(('Å', 'A', 'Ang', 'Angstrom', 'ang', 'angstrom'), 1.0e10, 1))
+    Q['10^-3 Angstrom^-1'] = 1e-3
+    DIMENSIONS['Q'] = Q
+
+    # Inverse volume units
+    scattering_volume = _build_inv_n_metric_units('meter', 'm', 3)
+    scattering_volume.update(_build_inv_n_units(('Å', 'A', 'Ang', 'Angstrom', 'ang', 'angstrom'), 1.0e10, 3))
+    DIMENSIONS['scattering_volume'] = scattering_volume
+
+    DIMENSIONS['SESANS'] = {'Å^{-2} cm^{-1}': 1, 'A^{-2} cm^{-1}': 1}
+
+    # Energy units
+    energy = _build_metric_units('electronvolt', 'eV')
+    DIMENSIONS['energy'] = energy
+    # Note: energy <=> wavelength <=> velocity requires a probe type
+
+    # Magnetic moment units
+    magnetism = _build_metric_units('tesla', 'T')
+    gauss = _build_metric_units('gauss', 'G')
+    gauss = dict((k, v * 1e-4) for k, v in gauss.items())
+    magnetism.update(gauss)
+    DIMENSIONS['magnetism'] = magnetism
+
+    # APS files may be using 'a.u.' for 'arbitrary units'.  Other
+    # facilities are leaving the units blank, using ??? or not even
+    # writing the units attributes.
+    unknown = {}  # type: Dict[str, ConversionType]
+    unknown.update(
+        {'None': 1, '???': 1, '': 1, 'A.U.': 1,  'a.u.': 1, 'arbitrary': 1, 'arbitrary units': 1,
+         'Counts': 1, 'counts': 1, 'Cts': 1, 'cts': 1, 'unitless': 1, 'unknown': 1, 'Unknown': 1, 'Unk': 1}
+    )
+    DIMENSIONS['dimensionless'] = unknown
+
+
+def standardize_units(unit: Union[str, None]) -> List[str]:
+    """
+    Convert supplied units to a standard format for maintainability
+    :param unit: Raw unit as supplied
+    :return: Unit with known, reduced values
+    """
+    # Convert value to a string -> Sets None to 'None'
+    # Useful for GUI elements that require string values
+    unit = str(unit)
+    # Catch ang, angstrom, ANG, ANGSTROM, and any capitalization in between
+    # Replace with 'Å'
+    unit = re.sub(r'[ÅAa]ng(str[oö]m)?(s)?', 'Å', unit, flags=re.IGNORECASE)
+    # Catch meter, metre, METER, METRE, and any capitalization in between
+    # Replace with 'm'
+    unit = re.sub(r'(met(er|re)(s)?)', 'm', unit, flags=re.IGNORECASE)
+    # Catch second, sec, SECOND, SEC, and any capitalization in between
+    # Replace with 's'
+    unit = re.sub(r'sec(ond)?(s)?', 's', unit, flags=re.IGNORECASE)
+    # Catch kelvin, KELVIN, and any capitalization in between
+    # Replace with 'K'
+    unit = re.sub(r'kel(vin)?(s)?', 'K', unit, flags=re.IGNORECASE)
+    # Catch celcius, CELCIUS, and any capitalization in between
+    # Replace with '℃'
+    unit = re.sub(r'cel(cius)?', '℃', unit)
+    # Catch hertz, HERTZ, hz, HZ, and any capitalization in between
+    # Replace with 'Hz'
+    unit = re.sub(r'h(ert)?z', 'Hz', unit)
```

Comment:
Is Hertz/Hz also allowed?

Reply:
think they need the ignorecase flags set